### PR TITLE
Disable medical AI for remote controlled AI and players

### PR DESCRIPTION
**Pull request by Glowbal**
**When merged this pull request will:**
- Fix medical AI forcing bandage even if the AI is remote controlled.

### CODE REVIEW

File: addons/medical_ai/StateMachine.hpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class GVAR(stateMachine) {
-    list = "allUnits select {local _x}";
+    list = "allUnits select {local _x && {_x != ACE_player} && {!isPlayer _x}}";
```
Comment (jonpas) [inline addons/medical_ai/StateMachine.hpp]: If it's not a player then it's also not ACE_player. Are both conditions really necessary?

Comment (PabstMirror) [inline addons/medical_ai/StateMachine.hpp]: Zeus remote controled AI return false for `isPlayer`
could also use `ace_common_fnc_isPlayer`

Comment (Glowbal) [inline addons/medical_ai/StateMachine.hpp]: if you are remote controlling a unit, your player unit is not `ACE_player` but your remote controlled AI is.

Comment (jonpas) [inline addons/medical_ai/StateMachine.hpp]: Ah, I see.

**Comment by esteldunedain** [comment]
I'm not sure this is enough.

 iirc CBA-FSM compiles a list of units to run, and then executes actions for one unit a time over the course of many frames, even seconds; when the list has been fully iterated then it is recalculated.

 What if you remote control a unit after the list was compiled but before that unit FSM is computed?

**Comment by BaerMitUmlaut** [comment]
This fix will not work. The state machine specifically runs *with* the players unit, so player will get treated by an AI medic when he's injured (for example in single player when you control a group of AI units).

Instead the fix should be applied here: https://github.com/acemod/ACE3/blob/master/addons/medical_ai/functions/fnc_healSelf.sqf#L16
The `ace_common_fnc_isPlayer` function PabstMirror mentioned should probably be used.

**Comment by BaerMitUmlaut** [comment]
Fixup magic! :shipit:

**Comment by BaerMitUmlaut** [comment]
Fixup magic! :shipit:

**Pull request merged**